Use new asset filter

tyleralsbury opened the pull request:
Refactor of SVGs in Dawn to use the new `inline_asset_content` Liquid filter.

tyleralsbury committed:
Moved snippets to assets and changed to svg

tyleralsbury committed:
Changed snippet icon rendering to use new inline_asset_content filter

tyleralsbury committed:
Removed hardcoded SVGs

tyleralsbury committed:
Fixed mistake in product media and refactored accordion icons

tyleralsbury committed:
Refactored inventory status icon

tyleralsbury committed:
Made SVGs valid

ludoboludo commented on snippets/cart-drawer.liquid:
```diff
@@ -478,7 +468,7 @@
                 {%- for discount in cart.cart_level_discount_applications -%}
                   <li class="discounts__discount discounts__discount--end">
-                    {%- render 'icon-discount' -%}
+                    {{ 'icon-discount.svg' | inline_asset_content }}
```
We should keep the white space control that was previously there.

ludoboludo commented on snippets/icon-with-text.liquid:
```diff
@@ -43,7 +43,7 @@
     <li class="icon-with-text__item">
       {%- if block.settings.image_1 == null -%}
-        {%- render 'icon-accordion', icon: block.settings.icon_1 -%}
+        {% render 'icon-accordion', icon: block.settings.icon_1 %}
```
I wonder if the whitespace control needed to be removed here 🤔

ludoboludo commented on snippets/pagination.liquid:
```diff
@@ -64,7 +64,7 @@
               aria-label="{{ 'general.pagination.next' | t }}"
             >
-              {%- render 'icon-caret' -%}
+              {{ 'icon-caret.svg' | inline_asset_content }}
```
whitespace control again

ludoboludo commented on snippets/product-media.liquid:
```diff
@@ -37,63 +39,80 @@
+  <button id="Deferred-Poster-Modal-{{ media.id }}" class="deferred-media__poster" type="button">
+    <span class="deferred-media__poster-button motion-reduce">
+      {%- if media.media_type == 'model' -%}
+        {{ 'icon-3d-model' | inline_asset_content }}
```
```suggestion
        {{ 'icon-3d-model.svg' | inline_asset_content }}
```

ludoboludo commented on snippets/product-media.liquid:
```diff
@@ -37,63 +39,80 @@
+      data-shopify-title="title"
+      data-shopify-xr-hidden
+    >
+      {{ 'icon-3d-model' | inline_asset_content }}
```
```suggestion
      {{ 'icon-3d-model.svg' | inline_asset_content }}
```

ludoboludo commented on snippets/product-thumbnail.liquid:
```diff
@@ -61,16 +61,14 @@
-            render 'icon-zoom'
-        endcase
-      -%}
+      {% case media.media_type %}
+        {% when 'video', 'external_video' %}
+          {{ 'icon-play.svg' | inline_asset_content }}
+        {% when 'model' %}
+          {{ 'icon-3d-model.svg' | inline_asset_content }}
+        {% else %}
+          {{ 'icon-zoom.svg' | inline_asset_content }}
+      {% endcase %}
```
```suggestion
      {% liquid 
        case media.media_type
          when 'video', 'external_video'
            echo 'icon-play.svg' | inline_asset_content
          when 'model'
            echo 'icon-3d-model.svg' | inline_asset_content 
          else
            echo 'icon-zoom.svg' | inline_asset_content
        endcase
      %}
```

ludoboludo commented on snippets/product-thumbnail.liquid:
```diff
@@ -106,10 +104,10 @@
         {%- if media.media_type == 'model' -%}
           <span class="visually-hidden">{{ 'products.product.media.play_model' | t }}</span>
-          {%- render 'icon-3d-model' -%}
+          {{ 'icon-3d-model.svg' | inline_asset_content }}
```
Ill do a whitespace specific check/commit for those.

ludoboludo committed:
address some of the comments I left

ludoboludo committed:
deal with whitespace to respect how it was before

tyleralsbury committed:
Fixed liquid error

tyleralsbury committed:
Merge branch 'use-new-asset-filter' of https://github.com/Shopify/dawn into use-new-asset-filter

melissaperreault commented on sections/main-product.liquid:
```diff
@@ -180,16 +178,14 @@
-                          <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(62,214,96, 0.3)"/>
-                          <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(62,214,96)"/>
-                        </svg>
+                        <span style="color: rgb(62,214,96, 0.3)">
```
```suggestion
                        <span style="color: rgb(62,214,96);display: flex;align-items: center;">
```

Adjusting here because:
- We need to center vertically the status circle
- Considering we can't apply an opacity to the outline, I suggest to remove the opacity entirely, I'll adapt the svg details on the file with my recommendations _(unless there is an alternative to keep the aesthetic of today, otherwise I am good.)_

tyleralsbury replied:
We can probably target it with CSS.  Lemme poke at it a bit.

melissaperreault replied:
Ok let me know! If it works, we can ignore my suggestion to change the svg details from this comment https://github.com/Shopify/dawn/pull/3582#discussion_r1715880411

melissaperreault replied:
Looks good color wise, just missing the vertical alignment to adjust! 🙌

tyleralsbury replied:
I will do that part in the Phase 3 PR because the way they're aligned changes there.

tyleralsbury committed:
Fixed inventory status

tyleralsbury commented on assets/icon-inventory-status.svg:
```diff
@@ -0,0 +1,4 @@
+<svg class="icon icon-inventory-status" width="15" height="15" aria-hidden="true">
+  <circle cx="7.5" cy="7.5" r="7.5" fill="currentColor" />
+  <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="currentColor" />
```
Fixed, it should match the same aesthetic as before.

melissaperreault commented on assets/icon-clipboard.svg:
Unsure if we are using 2 different clipboard icon on Dawn but this one is wrong for `Icon with text` block

| Current | Expected |
|--------|--------|
| ![](https://screenshot.click/13-35-ls4ct-37yx6.png) | ![](https://screenshot.click/13-38-6d1gn-hlon9.png) |

melissaperreault replied:
Ok, yes we got 2 versions and that is expected.

The clipboard one with thicker stroke is used for the [Share block copy link](https://screenshot.click/13-41-0tt40-1f5nm.png)

tyleralsbury committed:
Added proper accordion clipboard icon

melissaperreault commented on assets/square.svg:
I wonder if it is worth naming it checkbox? Just a detail

tyleralsbury replied:
It kinda just is a square... checkbox would imply a checkmark being involved, too, but this one really isn't anything but a square.

melissaperreault commented on sections/featured-collection.liquid:
```diff
@@ -188,7 +188,7 @@
     {%- endif -%}
     {% if section.settings.image_shape == 'arch' %}
-      {% render 'mask-arch' %}
+      {{ 'mask-arch.svg' | inline_asset_content }}
```
This doesn't apply the Arch shape to the product card images anymore.

| Current | Expected |
|--------|--------|
| ![](https://screenshot.click/14-49-6k9o3-uzaw1.png) | ![](https://screenshot.click/14-52-7ttlk-l7yl9.png) |

tyleralsbury replied:
That's weird.  There should be no changes to what's rendering.  I will investigate.

melissaperreault replied:
Yeah, I tried on the spin instance and it was not adjusting the shape either. Let's see! Thanks for taking a look!

tyleralsbury committed:
Refactor + Optimize svg files (#3586)

* Change width and height to be handled by parent context

* Optimized SVG files with svgo

tyleralsbury commented on assets/square.svg:
Fixed!  I musta done it in inspect element before.

tyleralsbury committed:
Fixes

bakura10 commented:
Is the inline_asset_content already in production Liquid? I tried to check how it works but it does not seem to work.

Remainder for the dev team as well: make sure to not forget app blocks and make it sure it works for assets inside an app block.

tyleralsbury commented:
> Is the inline_asset_content already in production Liquid? I tried to check how it works but it does not seem to work.
> 
> Remainder for the dev team as well: make sure to not forget app blocks and make it sure it works for assets inside an app block.

Not yet.  Keep an eye out, though.

tyleralsbury committed:
Fixed issues in header

melissaperreault commented:
**Header: Language select**

- [ ] Language selector is shorter in width, also wraps words where it didn't before
- [ ] The down caret is too far from the label

| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/23-10-0mmcr-5hepp.png) | ![](https://screenshot.click/23-08-zr5ha-0xsg7.png) |

melissaperreault commented:
**Mobile menu drawer**
- [ ] Arrows are not positioned correctly for nested menu
- [ ] The position of the menu close is not the same


| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/23-16-9w43f-6t2zd.png) | ![](https://screenshot.click/23-17-u68gw-urdrs.png) |

melissaperreault commented:
**Collection mobile filter**
- [ ] Icons are not aligned well to the right edge
- [ ] Deep a level, where checkboxes are, the label is not properly spaced and positioned next to the square
- [ ] Filter pills don't have the same size




| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/23-23-mh94k-br5l3.png) | ![](https://screenshot.click/23-25-fsf91-o6r6r.png) | 
| ![](https://screenshot.click/23-27-85060-leelj.png) | ![](https://screenshot.click/23-27-da1wq-c9f00.png) | 
| ![](https://screenshot.click/23-28-2ryp6-8tgk3.png) | ![](https://screenshot.click/23-29-optxo-rokd7.png) |

melissaperreault commented:
**Cart notification**

- [ ] Checkmark is smaller

| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/23-34-6hoc1-abvfp.png) | ![](https://screenshot.click/23-34-y0d35-j12ga.png) |

melissaperreault commented:
**Product template: Complementary block**

- [ ] When Quick add is activated, the Add to cart icon is the wrong one, should be a plus




| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/23-39-zrkgq-0v1hs.png) | ![](https://screenshot.click/23-39-k0xiz-i3spt.png ) |

melissaperreault commented:
**Quick order list**

- [ ] Error icon not aligned vertically ([Visual reference](https://screenshot.click/23-47-h11bv-z78lf.png))

tyleralsbury committed:
Batch of fixes

tyleralsbury commented:
@melissaperreault I sent a batch of fixes for the issues you found.  Great stuff!

tyleralsbury committed:
Fixed close icon

melissaperreault commented:
Thank you Tyler! 

I am investigating the active facets padding sizes to make them more accurate, stay tuned!

melissaperreault committed:
active facets padding adjustments

This is to make them shorter in height to stay closer to the original aesthetic.

melissaperreault commented on assets/component-facets.css:
```diff
@@ -582,7 +590,7 @@ span.active-facets__button-inner {
   min-height: 0;
   min-width: 0;
-  padding: 0.5rem 1rem;
+  padding: 0.25rem 1rem;
```
```suggestion
  padding: 0.25rem 1.25rem;
```

Also suggest to add this for a better horizontal white space

melissaperreault reviewed (commented):
**Drawer filter**
- [x] Drawer filter are not selectable, no checkmark ([Visual reference](https://screenshot.click/26-27-dwnfp-05duv.png))

tyleralsbury committed:
Fixed mobile and drawer facet checkbox

melissaperreault commented:
**Contact form**
- [ ] Success message icon is not vertically centered ([Visual](https://screenshot.click/26-49-ds04n-91rkm.png))

melissaperreault commented:
- [ ] Combining `Animation: Vertical lift` and `Product card: Image shape` setting cuts off the top of the shape ([Video reference](https://screenshot.click/26-04-z5y6t-dojgf.mp4))

melissaperreault commented:
**Drawer Filter checkmark**

- [ ] There is a clear difference in the thickness of the icon and the space it takes in the square. If we can't change the thickness, can we adjust the size of it?


| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/26-11-t4l5z-6mhgj.png) | ![](https://screenshot.click/26-13-d8hdj-qb4xz.png) |

melissaperreault commented:
Collecting this bug here but this exist on main.
- Complementary product block + Vertical lift/3D animation; there is an undesirable outline revealing on hover ([Video](https://screenshot.click/26-32-bf29u-qp5g4.mp4))

tyleralsbury commented on assets/base.css:
```diff
@@ -634,6 +634,14 @@ h3 .icon-arrow,
   transform: translateX(-0.05rem);
 }
 
+.svg-wrapper {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  width: 20px;
+  height: 20px;
+}
+
```
Good catch!

melissaperreault commented:
**Product cards x Bulk Quick add**

- [ ] View full details arrow is too small 
- [ ] The Volume pricing info icon is too small


| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/26-03-mz5d9-es1d1.png) | ![](https://screenshot.click/26-01-vtdro-jp4xk.png) |
|  <video src='https://github.com/user-attachments/assets/d0e103e5-7b58-4940-aff5-9a4cd14cb6cf' width=180/> | Cell |

melissaperreault commented:
**Cart drawer**

- [ ] Can't explain what I am seeing, but when I remove an item from cart, there is a shape revealing that we don't have on main.

| Expected | Current |
|--------|--------|
|  <video src='https://github.com/user-attachments/assets/d3a2e70c-99bd-4841-bfeb-05b55f8f884e'> | <video src='https://github.com/user-attachments/assets/bcdd8036-fb02-4e87-b35c-b21c737e460c'> |

tyleralsbury committed:
Another batch of fixes

tyleralsbury committed:
Adjusted quantity popover

tyleralsbury committed:
Fixed quantity info

tyleralsbury committed:
Fixed quantity info on desktop

melissaperreault commented:
👀 Found something very odd, some icons do not render on my iPhone, Safari browser. _*Note: Screenshot and browser version coming soon_

- [ ] Quantity inputs' minus and plus icon
- [ ] Login icon
- [ ] Close X on Cart Drawer, Filter drawer
- [ ] Icons from the `Icon with text` block from `Product template`
- [ ] Filter icon
- [ ] etc

melissaperreault commented:
**Complementary products block on Product information**

- [ ] When Quick add is enabled; Icons are not accurate between `Choose options` and `Add to cart`
- [ ] The `Plus` icon for `Add to cart` rotates, but not the appropriate way


https://github.com/user-attachments/assets/b811ea7a-a422-41fd-97d3-df6d457b9353

Roi-Arthur commented:
When using collection filters, the last checkbox seems slightly clipped. This seem to only be true if there are more than 2 options
[![](https://screenshot.click/27-57-3zsi3-2aa82.png)](https://screenshot.click/27-57-3zsi3-2aa82.png)


Unsure if this is directly linked to the recent changes since it shows up in the current version of Dawn, but some of the placeholder svg could use some extra CSS (`height: auto`)
[![](https://screenshot.click/27-47-3ji80-sb4zu.gif)](https://screenshot.click/27-47-3ji80-sb4zu.gif)

tyleralsbury committed:
Fix for Safari missing icons

tyleralsbury committed:
Fixed collapsible rows

melissaperreault commented:
A few regressions to note while fixing for Safari:

- [ ] Checkmark in Select is not vertically aligned ([Visual](https://screenshot.click/27-01-5qzim-x0aj8.png))
- [ ] Filter chevron not in place ([Visual](https://screenshot.click/27-21-cz9ob-feqyp.png))
- [ ] Complementary products block, when collapsible row turned on chevron is shifted ([Visual](https://screenshot.click/27-22-2woqq-zohje.png))

tyleralsbury commented:
> **Complementary products block on Product information**
> 
> * [x]  When Quick add is enabled; Icons are not accurate between `Choose options` and `Add to cart`
> * [x]  The `Plus` icon for `Add to cart` rotates, but not the appropriate way
> * [x]  Pagination numbers are not centered between the nav arrows ([Visual](https://screenshot.click/26-04-0ko0s-g053s.png))
> 
>  26-03-x2wka-sufej.mp4

This set was a bit weird - I didn't see these issues so it's possible another batch of fixes already fixed them.

tyleralsbury committed:
Batch of fixes

tyleralsbury committed:
Small batch of fixes

melissaperreault commented:
**Complementary products block**

- [ ] Size of the arrow is large

| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/27-40-l98ac-ypuxq.png) | ![](https://screenshot.click/27-41-x8cou-z8xx1.png) |

tyleralsbury committed:
Fixed Arthur's comments

tyleralsbury committed:
Another batch of fixes

melissaperreault commented:
**Gift card product**

- [ ] Checkmark is misplaced
- [ ] Error icon; Size of it and space between the icon and its label is wider

| Expected | Current |
|--------|--------|
| ![](https://screenshot.click/27-06-ztiq8-yz69m.png) | ![](https://screenshot.click/27-05-rb8xm-m8tu0.png) |

tyleralsbury committed:
Gift card fixes

tyleralsbury committed:
Nav fix

melissaperreault reviewed (approved):
Thank you Tyler and team for the hard work picking on the fine details. I am ready to put my approval on this! Let's continue to keep an eye on the other themes! 💪 🎉

Thibaut commented:
Congrats 👏

eugenekasimov reviewed (approved):
That's a huge amount of work 🚀

tyleralsbury commented:
> excuse me, where is this new filter documented?

The filter is not yet documented, but will be soon.

The usage is as seen in this Pull Request:

1. Add the file into the `/assets` folder.  Here we use it only for SVGs, but you can use other file types as well
2. In your Liquid `{{ 'file-name.ext' | inline_asset_content }}`

And that's all there is to it - it will inline the content of the file into the document.  Note that there is a filesize limit of 15kb for assets inlined by the filter.

tyleralsbury commented:
The docs are now available.

https://shopify.dev/docs/api/liquid/filters/inline_asset_content

montalvomiguelo commented:
There is an alternate approach to managing icons with SVG symbols, which is my favorite one, and I use it for my Tailwind projects. The icons could live in theme/assets/icons.svg instead of theme/snippets/icons.liquid. Then they can be referenced via <use> tag. 
```
<svg class="some essential classes">
  <use href="{{ icons.svg | asset_url }}#icon-arrow"></use>
</svg>
```

Since assets now load from the same origin, this approach works nicely. For the admin could use [svgxuse](https://github.com/Keyamoon/svgxuse).

PaulNewton commented:
`.quantity__button .svg-wrapper {`
https://github.com/Shopify/dawn/blob/main/sections/main-cart-items.liquid#L288

Troubleshooting breadcrumb: the quantity minus/plus has weird behavior when clicking the plus button it decrements, decreases, or minuses the qty; a merchants base.css may be the previous version but the button icon svg's have the span wrapper element with the `svg-wrapper` class for some reason in main-cart-items.liquid.
More commentary here https://github.com/Shopify/dawn/commit/755f6a448e25d40a8eb57550948ff3d074cf3818#r172904651
_, as I thought it would auto add to this thread /shrug_
